### PR TITLE
Add parameter parser

### PR DESCRIPTION
**Pull request by saralein**

### CODE REVIEW

File: server/src/main/java/com/saralein/server/parameters/ParameterParser.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        Map<String, String> params = new HashMap<>();
+        String[] splitUriAndParams = splitUriAndParams(uri);
+
+        if (splitUriAndParams.length > 1) {
+            String rawParams = splitUriAndParams[1];
+            params = formatParamsIfValid(rawParams);
+        }
+
+        return params;
+    }
```
Comment (zdavy) [inline server/src/main/java/com/saralein/server/parameters/ParameterParser.java]: 🤔 Do these exceptions make the server respond with an error code?

Comment (saralein) [reply server/src/main/java/com/saralein/server/parameters/ParameterParser.java]: Yeah.  Perhaps my PR is too small now and I should included the decoder and controller.

Comment (zdavy) [reply server/src/main/java/com/saralein/server/parameters/ParameterParser.java]: Yeah, it might help to see it all working together.

Comment (saralein) [reply server/src/main/java/com/saralein/server/parameters/ParameterParser.java]: OK, all the pieces should be there now.

**Review by zdavy — APPROVED** [review]

File: cobspec/src/main/java/com/saralein/cobspec/controller/ParameterController.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.List;
+import java.util.Map;
+
+public class ParameterController implements Controller {
+    private final ParameterParser parser;
+    private final ParameterDecoder decoder;
+
+    public ParameterController(
+            ParameterParser parameterParser,
+            ParameterDecoder parameterDecoder) {
```
Comment (damonkelley) [inline cobspec/src/main/java/com/saralein/cobspec/controller/ParameterController.java]: Do you think that the controller should be parsing and decoding the parameters? It feels like something you would want to do during the request parsing phase.

Comment (saralein) [reply cobspec/src/main/java/com/saralein/cobspec/controller/ParameterController.java]: I was going to make it a middleware.  But I didn't think about just moving it into the request parser.  Will look into moving it there.

**Pull request closed without merging**